src/jit/codegenarm64.cpp

```diff
@@ -450,6 +478,56 @@ void CodeGen::genBuildRegPairsStack(regMaskTP regsMask, ArrayStack<RegPair>* reg
         }
     }
     assert(regsCount == 0 && regsMask == RBM_NONE);
+
+    genSetUseSaveNextPairs(regStack);
+}
+
+//------------------------------------------------------------------------
+// genSetUseSaveNextPairs: Set useSaveNextPair for each RegPair on the stack which unwind info can be encoded as
+// save_next code.
+//
+// Arguments:
+//   regStack - a regStack instance to set useSaveNextPair.
+//
+// Notes:
+// We can use save_next for RegPair(N, N+1) only when we have sequence like (N-2, N-1), (N, N+1).
+// In this case in the prolog save_next for (N, N+1) refers to save_pair(N-2, N-1);
+// in the epilog the unwinder will search for the first save_pair (N-2, N-1)
+// and then go back to the first save_next (N, N+1) to restore it first.
+//
+// static
+void CodeGen::genSetUseSaveNextPairs(ArrayStack<RegPair>* regStack)
+{
+    for (int i = 1; i < regStack->Height(); ++i)
+    {
+        RegPair& curr = regStack->BottomRef(i);
+        RegPair  prev = regStack->Bottom(i - 1);
+
+        if (prev.reg2 == REG_NA || curr.reg2 == REG_NA)
+        {
+            continue;
+        }
+
+        if (genCanUseSaveNextPair(prev, curr))
+        {
+            curr.useSaveNextPair = true;
+        }
+    }
+}
+
+//------------------------------------------------------------------------
+// genCanUseSaveNextPair: Check that the curr regPair is adjusted to the next.
+//
+// Arguments:
+//   curr - a RegPair with 2 regs that we want to check;
+//   next - the next RegPair from the stack, also have to have 2 regs.
+//
+// static
+bool CodeGen::genCanUseSaveNextPair(RegPair curr, RegPair next)
```

Comment:
It's a little confusing here, perhaps, that in this function you use `curr` and `next` but in the function above that calls it, it is `prev` and `curr`.

You don't actually need a function here anymore, since it has only one call site.